tests/Integration/Database/EloquentModelDateCastingTest.php

```diff
@@ -35,6 +36,17 @@ public function test_dates_are_custom_castable()
         $this->assertInstanceOf(Carbon::class, $user->date_field);
         $this->assertInstanceOf(Carbon::class, $user->datetime_field);
     }
+
+    public function test_dates_are_diff_castable()
+    {
+        $user = TestModel1::create([
+            'datetime_diff_field' => '2019-10-01 10:15:20',
+        ]);
+
+        Carbon::setTestNow(Carbon::parse('2019-10-01 10:16:20'));
+
+        $this->assertEquals('1 minute ago', $user->toArray()['datetime_diff_field']);
+    }
```

Comment:
What happen if you try to save changes to casted value to database, since cast suppose to work both way.